src/grt/src/fastroute/src/utility.cpp

```diff
@@ -2495,8 +2495,13 @@ void FastRouteCore::saveCongestion(const int iter)
     const int seed = 42;
     g.seed(seed);
 
-    utl::shuffle(congestionGridsV.begin(), congestionGridsV.end(), g);
-    utl::shuffle(congestionGridsH.begin(), congestionGridsH.end(), g);
+    if (!congestionGridsH.empty()) {
+      utl::shuffle(congestionGridsH.begin(), congestionGridsH.end(), g);
+    }
```

Comment:
It would be better to make shuffle handle the empty vector correctly so that all users don't need this  check.

Reply:
`if (n == 0) return;` should do it

Reply:
Thanks! I'm testing it right now and will add it to the PR soon.